client/redux/users.js

```diff
@@ -17,11 +29,41 @@ export const getUsers = () => {
   }
 };
 
+export const destroyUser = (user) => {
+  return async (dispatch) => {
+    try {
+      await axios.delete(`/api/users/${user.id}`)
+      dispatch(_destroyUser(user))
+    } catch (err) {
+      console.log(err)
+    }
+  }
+}
+
+// Need to add the updateUser thunk
+export const updateUser = (user) => {
+  return async (dispatch) => {
+    try {
+      const updatedUser = await axios.put(`/api/users/${user.id}`, user)
+      console.log("updated", updatedUser)
+      dispatch(_updateUser(user))
+    } catch (err) {
+      console.log(err)
+    }
+  }
+}
+
+
+
 export default function usersReducer(state = [], action) {
   switch (action.type) {
     case GET_USERS:
       return action.users;
+    case DESTROY_USER:
+      return state.filter(user => user.id !== action.user.id)
+    case UPDATE_USER:
+      return action.user
```

Comment:
I don't think the reducer should work this way for UPDATE

Reply:
Done